dotCMS/src/test/java/com/dotmarketing/filters/CMSUrlUtilTest.java

```diff
@@ -0,0 +1,31 @@
+package com.dotmarketing.filters;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.UnsupportedEncodingException;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.Test;
+
+/**
+ * @author nollymar 
+ */
+public class CMSUrlUtilTest {
+
+    /**
+     * Method To Test: {@link CMSUrlUtil#getURIFromRequest(HttpServletRequest)}
+     * Given Scenario: Call {@link CMSUrlUtil#getURIFromRequest(HttpServletRequest)}
+     * with a request that contains a mocked URI with a plus sign
+     * ExpectedResult: The URI returned must be the same one mocked,
+     * including the plus sign
+     */
+    @Test
+    public void testGetURIFromRequest() throws UnsupportedEncodingException {
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        when(request.getRequestURI()).thenReturn("dotcms+test.txt");
+        String result = CMSUrlUtil.getInstance().getURIFromRequest(request);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'result' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4763071)

Reply:
Done